### DISCUSSION TITLE
OIDC with AWS Cognito

### DISCUSSION THREAD

**Discussion by thirsty-bankroll** [Budibase/budibase#6689] [Help]
I tried to connect the open id connect plugin with aws cognito 

My config url is : https://cognito-idp.{region}.amazonaws.com/{userPoolId}/.well-known/openid-configuration
Budibase callback : https://testbudieri.budibase.app/api/global/auth/testbudieri/oidc/callback 
Budibase app : https://testbudieri.budibase.app/app/school-admin-panel

On the main page, I put a login button to redirect to the cognito login UI, but when I login, I don't know how to handle this error:

{"message":"Error constructing OIDC authentication configuration","status":500}

![image](https://user-images.githubusercontent.com/23150157/178781330-cde49a50-d898-4cba-b31d-45cb319d188b.png)

**Comment by thirsty-bankroll** [comment]
Anyone ?

**Comment by aptkingston** [comment]
Hey @d22ier. Our OIDC integration is designed to be used for logging in to Budibase, not as a custom implementation within an app. Providing our OIDC callback URL to cognito will not work unless you've correctly set up cognito as your identity provider in your Budibase auth settings. Your client apps should not be concerned with logging in and out - that's the responsibility of the  Budibase platform.

e.g. configure Cognito as your OIDC provider:
<img width="1800" alt="image" src="https://user-images.githubusercontent.com/9075550/179566197-0ffff947-5148-427f-9dd8-b732a0b3d322.png">

Then you can log in using Cognito as your identity provider via OIDC:
<img width="599" alt="image" src="https://user-images.githubusercontent.com/9075550/179566348-0e7aa7fb-93cc-4451-9067-1d570975ccee.png">

Having said all this... I tried this myself and was unable to get Cognito to work properly. Maybe you'll have more luck than I did, as I'm not particularly familiar with Cognito setup.
